input/otlp/traces.go

```diff
@@ -1056,3 +1057,55 @@ func schemeDefaultPort(scheme string) int {
 	}
 	return 0
 }
+
+// parses traceparent header, which is expected to be in the W3C Trace-Context
+// and searches for the p-value as specified in
+//
+//	https://opentelemetry.io/docs/reference/specification/trace/tracestate-probability-sampling/#p-value
+//
+// to calculate the adjusted count (i.e. representative count)
+func getRepresentativeCountFromTracestateHeader(tracestace string) float64 {
+	var p uint64 = 0
+
+	otValue := getValueForKeyInString(tracestace, "ot", ',', '=')
+
+	if otValue != "" {
+		pValue := getValueForKeyInString(otValue, "p", ';', ':')
+
+		if pValue != "" {
+			p, _ = strconv.ParseUint(pValue, 10, 6)
```

Comment:
The lint issue below raises a new question for me: if the "p" key is missing or invalid from tracestate, then `p` here will be zero. That leads to a representative count of 1 (2**0). Is that intentional? I guess that's the right default, since otherwise there should not have been a span event sent. Worth adding a comment?

Reply:
> it's not making it worse to the previous situation, right?

Agreed, thanks for the update